### PR TITLE
Add /create_scenario endpoint and updates associated models

### PR DESCRIPTION
**Pull request by elsiel23**
This endpoint should be called after the user clicks “Generate Scenario” in the Plan flow if weights are provided. 
- Project is not a required FK as discussed. We'll set constraint values during object creation via the front end.
- Specific errors: if weights aren't provided, if priorities aren't provided, if len(weights) != len(pri)

Example input:
{'plan_id': self.plan.pk,
  'priorities': ['cond'],
  'weights': [1],
  'max_budget': 100
}

### CODE REVIEW

**Comment by elsiel23** [comment]
@thequeenofspades @lattemilk FYI

File: src/planscape/plan/models.py [hunk 1]
```diff
@@ -69,9 +69,6 @@ class Project(models.Model):
     # Ratio of elevation to distance
     max_slope: models.FloatField = models.FloatField(null=True)
 
-    # TODO: Add more project parameters like min_acres_treated and
-    # permitted_ownership = (1=federal, 2=state, 4=private)
-
 
 class Scenario(models.Model):
```
Comment (strangeties) [inline src/planscape/plan/models.py:73]: just to double-check: does this correspond with RankingScenario in the db graph?

File: src/planscape/plan/models.py [hunk 2]
```diff
@@ -82,16 +79,39 @@ class Scenario(models.Model):
...
     owner = models.ForeignKey(
         User, on_delete=models.CASCADE, null=True)  # type: ignore
+    
+    plan = models.ForeignKey(Plan, on_delete=models.CASCADE, null=True)  # type: ignore
+
+    # The creation time of the project, automatically set when the project is created.
+    creation_time: models.DateTimeField = models.DateTimeField(
+        null=True, auto_now_add=True)
 
     project = models.ForeignKey(
```
Comment (strangeties) [inline src/planscape/plan/models.py:89]: Again to double-check: does Project correspond to ForsysParams in the db graph?

File: src/planscape/plan/views.py [hunk 3]
```diff
@@ -425,6 +434,73 @@ def get_project_areas(request: HttpRequest) -> HttpResponse:
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
 
 
+def _save_scenario_metadata(max_budget, max_treatment_area_ratio, max_road_distance, max_slope, priorities, weights, scenario: Scenario):
```
Comment (strangeties) [inline src/planscape/plan/views.py:437]: nit: the auto-formatter should be able to split this into multiple lines.

**Comment by strangeties** [comment]
Looks fine!

Just a broader thought: I think this pr follows an earlier DB design, which conflates project area generation scenarios with project area ranking scenarios. Is that the intention?

File: src/planscape/plan/models.py (hunk 1 above)
Comment (elsiel23) [reply src/planscape/plan/models.py:73]: yes!

File: src/planscape/plan/models.py (hunk 2 above)
Comment (elsiel23) [reply src/planscape/plan/models.py:89]: yes mostly, although I'm not sure if that's changing given our discussion today. should we chat about it?

File: src/planscape/plan/views.py [hunk 4]
```diff
@@ -185,31 +185,37 @@ def list_plans_by_owner(request: HttpRequest) -> HttpResponse:
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
 
 
-def _save_project_parameters(body, project: Project):
-    # Parse constraints
-    max_budget = body.get('max_budget', None)
+def _validate_constraint_values(max_budget, max_treatment_area_ratio, max_road_distance, max_slope):
```
[the remark this answers is not in the document]
Comment (elsiel23) [reply src/planscape/plan/views.py:188]: is it ok if neither is set for project area ranking? i think scenario/rankedscenario is a shared object between  the two flows

File: src/planscape/plan/views.py (hunk 3 above)
Comment (elsiel23) [reply src/planscape/plan/views.py:437]: 1: weird! it doesn't catch it, but fixed manually
2: done thanks!

File: src/planscape/plan/models.py (hunk 2 above)
Comment (strangeties) [reply src/planscape/plan/models.py:89]: lol. May we chat about it after the product vision for configs and scenarios has settled more?

(though if there's a sense we can get consensus between us and can convince everyone else it's the right way forward, maybe we can chat about it first?)

In the meantime, I don't want to block interfacing work and will go with what you're implementing.

**Comment by elsiel23** [comment]
re: "conflates project area generation scenarios with project area ranking scenarios" Maybe we're still not on the same page about this. Could you help me understand which tables represent a project area generation scenario/outut vs a ranking scenario/output? I'm using scenario to mean the output the user sees at the end of a single forsys run (patchmax or non patchmax) which includes ranking.

File: src/planscape/plan/views.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (strangeties) [reply src/planscape/plan/views.py:188]: I see! yea - it's fine if neither is set for ranking.

Is it possible to check whether a scenario is for ranking or generation? If so, can you add logic for validating that at least one of area and budget constraints is set? If not, can you add a TODO?

**Comment by strangeties** [comment]
After today's discussion, I have better sense of what everyone else is thinking regarding scenarios, etc. - I should redact that last comment about scenarios because I was defining scenario as an instance of priority weights, not as the entire input of a forsys run.

File: src/planscape/plan/views.py (hunk 3 above)
Comment (strangeties) [reply src/planscape/plan/views.py:437]: thanks!
I don't see this in the code yet but I assume it's a matter of pushing the local repo.

**Review by strangeties — APPROVED** [review]

File: src/planscape/plan/migrations/0012_scenario_creation_time_scenario_max_budget_and_more.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('conditions', '0004_get_condition_pixels'),
+        ('plan', '0011_project_creation_time'),
+    ]
+
+    operations = [
+        migrations.AddField(
```
Comment (lattemilk) [inline src/planscape/plan/migrations/0012_scenario_creation_time_scenario_max_budget_and_more.py:15]: double checking here - is there also a field for the creator's notes (in the mocks)?

File: src/planscape/plan/models.py (hunk 1 above)
Comment (elsiel23) [reply src/planscape/plan/models.py:73]: leaving these changes as is for now and we'll revise again with new DB adjustments

File: src/planscape/plan/migrations/0012_scenario_creation_time_scenario_max_budget_and_more.py (hunk 5 above)
Comment (elsiel23) [reply src/planscape/plan/migrations/0012_scenario_creation_time_scenario_max_budget_and_more.py:15]: i haven't added that yet, let work on this in a follow up

**Pull request merged**